feat: Add Assay profile and example

bwalsh opened the pull request:
# Motivation

This PR addresses #50 

```
How can we track informatics assays within a FHIR schema to ensure they are findable and searchable (using REST API - search parameters of a resource), while maintaining strong associations (tightest links) with core FHIR resources such as Specimen, DocumentReference, Patient, and Observations?
```
# Approach
updated 7/30/25  -  see [documentation](https://github.com/NIH-NCPI/ncpi-fhir-ig-2/blob/feature/assay-servicerequest/input/pagecontent/module_assays.md) 
 
<img width="866" height="601" alt="image" src="https://github.com/NIH-NCPI/ncpi-fhir-ig-2/blob/feature/assay-servicerequest/input/images/assay-usage.png?raw=true" />

awarkow committed:
:memo:

awarkow committed:
:memo: fix pedigree link

awarkow committed:
✏️ fix typos

bwalsh committed:
feat: Add Assay profile and example

bwalsh committed:
feat: Add Assay profile and example

teslajoy reviewed (dismissed):
Love the isolation of Task DAG - this will be immensely helpful for orchestrating ETL pipelines. Two minor points (as we’ve noted before):

1. LOINC coverage for bioinformatics workflows is still patchier than EBI’s UBERON ontology.
2. Open-source consortium datasets rarely expose true longitudinal DateTime events; you could bake a composite identifier - combining an index-date days  (anchor point) with the submitter ID to encode longitudinal context. The de-identified time definations are not a new concept and can be adapted from resources like GDC etc. 

_neither are required here... highly appreciate it. Just making a note._ 

Otherwise, goodness this looks solid! Thank you kindly Walsh!

torstees committed:
Fixed an error relating to the assay references

torstees committed:
Fixed an error relating to the assay references

torstees reviewed (changes requested):
There are a couple of [new errors](https://deploy-preview-126--ncpi-fhir-ig-v2.netlify.app/qa.html#_home_runner_work_ncpi-fhir-ig-2_ncpi-fhir-ig-2_fsh-generated_resources_ServiceRequest-a1) in the qa report that seem to be related to assays. Would you look into those and see if they are fixable?

bwalsh commented:
@torstees - thanks.  will look into these today

bwalsh committed:
removes commented; adds ncit

bwalsh committed:
adds ncit

bwalsh commented:
@torstees - fixed  IG generation errors (noted that fsh-generated/fsh-index.json -> fsh-generated/data/fsh-index.json)
@teslajoy  - added ncit ontology

bwalsh committed:
move AgeAtAssertion

bwalsh committed:
improve age-at-assertation example

bwalsh committed:
improve age-at-assertation example

torstees committed:
Update R4 with R4B (#99)

torstees committed:
Revert Back to R4 (#138)

* trying to remove them once again

* Remove machine generated files

* code systems and valuesets we depend on that aren't present in R4

* Revert to R4

* Fixed system warning

* CS/VS warning cleanup

* CS/VS warning cleanup

* various warnings and errors

* CS/VS warning cleanup

* Various warning and error tweaks

* Add new systems and fixed the MeSH System

* Cleaned up a number of broken links

* Update documentation to describe how to link to papers now that we no longer have the Citation resource

* shouldn't have added this in

* R4B Cleanup

* Seeing if the mcode version change is causing issues with GHA

* correct version of mcode

* Added placeholder and link to make finding the fsh-generated index easier to find.

* Header/heading changes to make more consistent with the rest of the IG

* Removed this since it wasn't helping with the menu link

RobertJCarroll committed:
🐛 Remove constraints on ids and identifiers

RobertJCarroll committed:
🐛 Add some id/identifier back in

I pruned too heavily, so I put a couple of ids back.

bwalsh committed:
Correct tense

bwalsh commented:
@torstees @RobertJCarroll
I've rebased and corrected AgeAtAssertion tense

<img width="1043" height="103" alt="image" src="https://github.com/user-attachments/assets/eeae27fe-3ca9-4034-9759-b48ba25ddb4d" />


I believe these are the same errors in the [released version](file:///Users/walsbr/ncpi/ncpi-fhir-ig-2/output/qa.html)

<img width="1206" height="220" alt="image" src="https://github.com/user-attachments/assets/c1d00652-9b71-4b10-adab-f2c5be3201ab" />

RobertJCarroll commented:
I was seeing some of those errors myself working in #145 , I didn't investigate further.

RobertJCarroll commented:
Re the content: I see you have this configured as as ServiceRequest. That's a `Request` type per this [Workflow ](https://hl7.org/fhir/R4/workflow.html)page. Would it make sense to use one of the `Event` type resources here given we are expecting only to get reported results? I see you mention Tasks, but I don't see it in the diagram. I'm not sure how the NCPI Informatics Assay Request would like directly to an output file.

torstees commented on input/fsh/modules/assay.fsh:
```diff
@@ -0,0 +1,12 @@
+* status 1..1
+* intent = #order
+* code 1..1 MS
+* code from $loinc (preferred)
```
If you want to specify a Code System, this should be * code from system $loinc (preferred)   According to the qa report, what you have there is expected to be a Value Set.

torstees reviewed (changes requested):
I pointed out an issue with your NcpiAssay profile. I think you need system inside that binding since you are pointing to a code system instead of a valueset.

bwalsh committed:
review feedback

bwalsh committed:
explicit application descriptions

bwalsh committed:
update documentation

bwalsh committed:
update documentation

bwalsh commented on input/fsh/modules/assay.fsh:
```diff
@@ -0,0 +1,17 @@
+// $loinc (preferred): This keeps the existing binding to LOINC as the preferred code system.
+// $ncit: Adds NCIT as an another code system, allowing codes from NCIT to be used in addition.
+* code from $loinc (preferred)
+* code from $ncit (preferred)
```
@torstees Thank you.  It finally sank in what you were trying to say.   I've leveraged Device/DeviceDefinition to allow the author explicitly specify what performed the assay

bwalsh replied:
deprecated

bwalsh commented:
> Re the content: I see you have this configured as as ServiceRequest. That's a `Request` type per this [Workflow ](https://hl7.org/fhir/R4/workflow.html)page. Would it make sense to use one of the `Event` type resources here given we are expecting only to get reported results? I see you mention Tasks, but I don't see it in the diagram. I'm not sure how the NCPI Informatics Assay Request would like directly to an output file.

@RobertJCarroll 
 

Thank you for the feedback!

We chose `ServiceRequest` to represent the **assay intent**, aligning with HL7's [[Workflow](https://hl7.org/fhir/R4/workflow.html)](https://hl7.org/fhir/R4/workflow.html) pattern of `Request → Event → Result`.

* The `Task` resource captures the **execution event**, and is linked to the `ServiceRequest` via `Task.basedOn`.
* Resulting files are represented using `NcpiFile`, tied to either the `ServiceRequest.context.related` or `Task` context.

To support **computational provenance**, we include:

* **`DeviceDefinition`**: Describes the **software assay** (e.g., BWA, GATK) — including version, source, and manufacturer.
* **`Device`**: Represents the **specific instance** of that software used during the execution, and is referenced in `Task.performer`.

This structure enables a complete and traceable path from assay request to execution to resulting output files. I've updated the diagram to make this clearer.

Also, if no specific provenance is known, a default might be "Submitted Files Intake" to represent de-facto uploaded files.

RobertJCarroll commented:
I think we might be talking past each other a little on what we mean, I'll try and clarify.

> We are operating in a retrospective reporting model -> Not all datasets are. Current ETL data builds can’t map due to lack of schema model standards.

The framing of the IG is something along the lines of: "We are cloud-based data repositories are providing access to existing shared datasets". That's what I mean by retrospective- NCPI FHIR as a whole isn't operating a "live" system where requests to run collect specimens, run assays, or apply workflows happen. We are focused on sharing what has already happened- ie, reporting the specimens that were collected and which assays and workflows were run. If a platform is doing those activities awesome, but I believe that's outside the scope of where we are today.
 
> Why we can’t include all of the information in a Task -> The issue stems from how FHIR models DocumentReference, Specimen, and Patient relationships across versions. Task doesn’t explicitly address these relations, it’s designed to capture provenance.

I think you may be making some distinction here that I'm unfamiliar with, so please let me know if I'm missing something. I am expecting that `Task.input` and `Task.output` cover those relations, eg, what samples came in and what files came out. Slicing them to provide the correct expected `type`s and `value[x]`s is the role of the profile here. That's present in R4 and R5 in the same way.

> Not familiar with a dataset providing that level of detail -> If we focus on relations and provenance, entities matter more than fields. A simple version of provenance across file, specimen, and patient over time can't fully be captured with current standard and different models.

It sounds like you have a concrete use case here that I'm unfamiliar with. I agree that we don't have any explicit ability to capture multiple `Patient`/participant snapshots over time, though this is something we could discuss for the future. My comment was specifically saying "I'm not familiar with any shared dataset that explicitly provides the details of the assay that was requested by the researcher, eg, documenting the order they sent to a sequencing center, which is the information we'd need to accurately populate a `Request` type resource. All of the datasets I've seen only provide information about what happened, which doesn't fit the precise semantics of `Request` type resource.

> Keeping them distinct avoids conflating process history with structural or assay file-based relationships. I'm just surfacing these points in case they help clarify things.

I'm on board with providing/recognizing avenues to capture all of these elements! My concern is that `Request` type resources do not record what actually happened, so we would still need to ensure all of these elements can be captured in an `Event` type resource as well. Pulling the descriptions from [workflow](https://hl7.org/fhir/R4/workflow.html):
>Definitions	Resources that define something that can potentially happen in a patient and time-independent manner
>Requests	Resources that ask for or express a desire/intention for something to be done
>Events	Resources that express that something has been done and which can potentially be done because of a request

bwalsh committed:
Adds DiagnosticReport, ActivityDefinition

bwalsh commented:
# Structuring Genomic Assays in FHIR: The HTAN Perspective

## Summary
This document outlines the **HTAN Genomic Assay Model** in FHIR,
detailing how genomic assays are structured, ordered, and reported within the FHIR framework.

It introduces the `DiagnosticReport`, as the `Event` -typed resource in the FHIR workflow, which encapsulates the assay results and their interpretations.


## HTAN: Harmonized Tissue Architecture Network
HTAN is a **retrospective, multi-institutional data repository** with contributions spanning **diverse platforms and deeply layered research studies**. These studies address different biological questions but together form a **complementary, interoperable foundation**, connecting datasets across **assays, timepoints, and institutions**.

Data is submitted as **derived files**, each anchored to a specific **assay**, **specimen**, and **defined timepoint**:

- **Files** are the primary unit of exchange and curation  
- **Assays** are explicitly annotated by workflow and level, and are not inferred  
- **Devices and platforms** are essential to interpreting assay results and file context  
- **Temporality** (e.g., longitudinal sampling, time from index) is embedded in metadata and increasingly standardized as de-identified “days to” intervals across institutions  

This structure enables scalable, harmonized analyses.

bwalsh committed:
Adds DiagnosticReport, ActivityDefinition

bwalsh committed:
Measurement hierarchy

bwalsh committed:
adds query examples

bwalsh committed:
rm AgeAtAssertion

bwalsh committed:
Adds Age extension

bwalsh committed:
Adds EDAM

bwalsh committed:
update diagram

bwalsh committed:
update doc

bwalsh commented:
Adding some notes from slack discussion


---

### 🧬 Thoughts on ServiceRequest, Assay Modeling, and the Path Forward

Thanks for the great commentary — there’s a healthy tension here between **modeling parsimony** and **semantic completeness**. I’d like to affirm and extend some of the key points raised.

---

#### ✅ Why Keep `ServiceRequest`?

You're absolutely right that **ServiceRequest feels duplicative** in many retrospective settings. However, if we want to align with **FHIR’s canonical workflow pattern**, it's worth noting:

> 🔗 `ServiceRequest` is the **only resource that links to `ActivityDefinition`** in both R4 and R5.

This makes it essential for use cases where we want to:

* Describe what was intended or planned (even retrospectively, for harmonization)
* Group procedures by assay type (e.g., "all scRNA-seq v3 assays")
* Trace standard workflows or protocols

In HTAN-like datasets, even if we didn’t capture the request at the time, modeling it post hoc gives us a **handle for harmonization across diverse pipelines**.

---

#### 🧪 `Procedure`, `DiagnosticReport`, and Generalizing Beyond Genomics

> *“What if we just use `Procedure`?”*

That’s a valid path — and **nothing precludes us from including `Procedure`** when describing real-world events like biopsies, dissection, extraction, or sequencing runs. In fact, in a complete model, we likely *should*.

However, relying solely on `Procedure` or `DiagnosticReport` risks overfitting if we’re not careful:

* `DiagnosticReport` in R5 makes strong assumptions about **reference types**, especially toward `GenomicStudy`.
* By sticking with "stock" FHIR patterns — `ActivityDefinition`, `ServiceRequest`, `Procedure`, `Task`, `Observation`, `DocumentReference` — we avoid overcommitting to genomics-specific resources, making it easier to represent proteomics, metabolomics, spatial assays, etc.

---

#### 🧱 `Task` as a Lightweight Model

> *“Maybe we could just use Task?”*

Yes, `Task` is appealing in its flexibility:

* It captures **parameters**, **inputs**, and **outputs** clearly
* It reflects **real-world execution steps**, including system-level orchestration

But there's a catch:

> ❗ `Task` **does not link cleanly** to `Specimen`, `Group`, or `ActivityDefinition`.

You could overload `Task.basedOn` with references to those entities, but it’s not semantically clean. You also lose the formal link back to reusable workflows (ActivityDefinition), which breaks alignment with the FHIR workflow pattern.

---

### 📊 Comparison of Two Modeling Strategies

| Aspect                                         | **FHIR Workflow Model**<br>(ActivityDefinition → ServiceRequest → DiagnosticReport → DocumentReference) | **Abbreviated Model**<br>(Task + DocumentReference)                      |
| ---------------------------------------------- | ------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------ |
| 🔄 **Workflow Intent**                         | ✅ Explicit via `ActivityDefinition` and `ServiceRequest`                                                | 🚫 No clear modeling of intent or plan                                   |
| 📄 **File Representation**                     | ✅ Via `DiagnosticReport.media`                                               | ✅ Via `Task.output` referencing `DocumentReference`                      |
| 🧫 **Specimen Linkage**                        | ✅ Supported throughout (SR, DR, Procedure, Obs)                                                         | 🚫 No direct `specimen` element in `Task`; workaround via `Task.basedOn` |
| 🔬 **Procedure Representation**                | ✅ Full fidelity via `Procedure`, optionally linked to SR                                                | 🚫 Lacks support for real-world clinical procedures                      |
| 🔁 **Reuse of Standard Workflows**             | ✅ Via `ActivityDefinition`                                                                              | 🚫 No link to reusable protocol definitions                              |
| 🧬 **Support for Genomics & Beyond**           | ✅ Generalizable if `GenomicStudy` is optional                                                           | ✅ Generalizable, but at cost of semantic richness                        |
| 🧰 **Support for Parameters, Runtime Context** | 🟡 Partial (mostly in Task, some in SR)                                                                 | ✅ Strong via `Task.input`/`output`                                       |
| ⚙️ **Execution Lifecycle**                     | 🟡 Implicit                                                                                             | ✅ Explicit (via Task.status, Task.requester, etc.)                       |
| 🧵 **Traceability Across Resources**           | ✅ Rich, bidirectional links                                                                             | 🚫 Flat, no links between workflow entities                              |

---

### ✅ Summary

* `ServiceRequest` is optional **from a data completeness standpoint**, but **required if you want to tie to `ActivityDefinition`**.
* Including `Procedure` is encouraged for real-world assay execution steps (e.g., biopsy, extraction).
* `Task` is great for execution and parameter tracking — but lacks key clinical linkages unless overloaded.
* A combined approach may work best:
  * use simple `DocumentReference` and `DocumentReference.subject` when representing submissions without an assay,
  * but supplement with workflow modeling and retrospective harmonization.

Let’s keep iterating toward a flexible, modular pattern that serves both **prospective workflows** and **retrospective analysis**.


---

## 🧭 Observation: `DocumentReference.basedOn` in R5 May Supersede the Need for `DiagnosticReport`

### ✅ Key Insight

In **FHIR R5**, [`DocumentReference.basedOn`](https://hl7.org/fhir/documentreference.html#DocumentReference.basedOn) was added to allow direct linkage to the **request** that led to the creation of the document — typically a `ServiceRequest`.

> This allows a `DocumentReference` to stand alone as a traceable outcome of an assay **without requiring** a `DiagnosticReport`.

---

### 🔗 Implications for Assay Modeling

| In R4                                                                                                                                       | In R5                                                                                 |
| ------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------- |
| `DiagnosticReport` is often required to mediate between `ServiceRequest`, `Procedure`, `Observation`, and the actual document (via `media`) | `DocumentReference.basedOn` can directly fulfill the "result of request" relationship |
| To represent assay output (e.g. a FASTQ file), you need: `ServiceRequest → DiagnosticReport → media.extension → DocumentReference`          | You can simplify to: `ServiceRequest → DocumentReference.basedOn`                     |

---

### 🛠️ Modeling Simplicity in R5

This change means that in **R5**, for many real-world retrospective assays, **you may no longer need `DiagnosticReport` at all**, especially if:

* You’re using `DocumentReference` to point to structured or unstructured outputs (e.g. PDFs, BAMs, imaging data, etc.)
* You have no additional clinical interpretation metadata that would warrant use of `DiagnosticReport` (e.g. status, conclusion, coded results, performer roles)

---

### 📌 Caveat: Loss of Semantic Richness

While this shortcut improves simplicity, dropping `DiagnosticReport` means:

* You lose structured **clinical interpretation fields** (e.g., `conclusionCode`, `presentedForm`, etc.)
* You forgo **result grouping** across multiple Observations (e.g., variant calls, expression data)
* You remove a formal **event resource** that could anchor provenance and orchestration semantics

---

### 🧬 Summary

| Modeling Strategy                                          | Verbosity    | Traceability | Clinical Richness                                   |
| ---------------------------------------------------------- | ------------ | ------------ | --------------------------------------------------- |
| `ServiceRequest → DiagnosticReport → DocumentReference`    | More verbose | High         | High                                                |
| `ServiceRequest → DocumentReference` (via `basedOn` in R5) | Simpler      | Medium       | Lower (unless coded in `DocumentReference.content`) |

---

If you're building an **execution-focused assay registry** or **data delivery pipeline**, R5's `DocumentReference.basedOn` could dramatically streamline modeling.

But if you're building **clinically interpretable bundles** or **longitudinal assay tracking**, `DiagnosticReport` still offers substantial value.



# 🔗 FHIR R4 Extension: `diagnosticreport-media-documentreference`

This extension allows a `DiagnosticReport.media` element in **FHIR R4** to reference a `DocumentReference` resource instead of a `Media` resource—mirroring **FHIR R5** behavior.

---

## 🧩 Purpose

In FHIR R4:
- `DiagnosticReport.media.reference` only accepts `Reference(Media)`
- FHIR R5 expands this to include `Reference(DocumentReference)`

This extension bridges that gap, enabling forward-compatible modeling of rich external documents (e.g., PDFs, structured genomic reports).

---

## ✅ Pros

- **Aligns with FHIR R5**: Simplifies version migration and shared tooling across R4 and R5.
- **Enables Document Modeling**: Use `DocumentReference` to represent PDFs, DICOM, structured outputs, etc.
- **Rich Metadata**: Includes authorship, custodians, access control, and external storage links (e.g., DRS/S3).
- **Bidirectional Navigation**: Complements `DocumentReference.context → DiagnosticReport`.

---

## ❌ Cons

- **Not Valid in Base R4**: Requires custom profiles and StructureDefinitions.
- **Limited Tooling Support**: Most FHIR R4 tools expect `media.reference` → `Media`.
- **Extension Overhead**: Must be defined and consistently used across systems.

---

## 📦 FSH: Extension Definition

```fsh
Extension: DiagnosticReportMediaDocumentReference
Id: diagnosticreport-media-documentreference
Title: "DiagnosticReport Media DocumentReference"
Description: "Enables a DiagnosticReport.media entry to reference a DocumentReference instead of a Media resource, aligning with FHIR R5 behavior."
* ^status = #active
* ^context[+].type = #element
* ^context[=].expression = "DiagnosticReport.media"
* value[x] only Reference(DocumentReference)
* valueReference ^short = "A DocumentReference representing the media content (e.g., PDF, image, structured report)."
```

---
## 📦 FSH: SearchParameter Definition

```fsh
Profile: SPDiagnosticReportMediaDocumentReference
Parent: SearchParameter
Id: sp-diagnosticreport-media-documentreference
Title: "SearchParameter for DiagnosticReport.media.extension:DocumentReference"
Description: "Enables _include and _revinclude traversal from DiagnosticReport to DocumentReference and vice versa using the diagnosticreport-media-documentreference extension."

* name = "media-documentreference"
* status = #active
* code = "media-documentreference"
* base = #DiagnosticReport
* type = #reference
* expression = "DiagnosticReport.media.extension.where(url='http://example.org/fhir/StructureDefinition/diagnosticreport-media-documentreference').value as Reference"
* target = #DocumentReference
* url = "http://example.org/fhir/SearchParameter/diagnosticreport-media-documentreference"
* xpathUsage = #normal
* description = "Reference to a DocumentReference resource used in DiagnosticReport.media via custom extension."

```

### 🔍 Usage Examples
➕ _include from DiagnosticReport → DocumentReference
```
GET /fhir/DiagnosticReport?_include=DiagnosticReport:media-documentreference
```
🔁 _revinclude from DocumentReference → DiagnosticReport

```
GET /fhir/DocumentReference?_revinclude=DiagnosticReport:media-documentreference
```


---

## FSH: Usage Example

```fsh
Instance: example-report
InstanceOf: DiagnosticReport
Title: "Genomic Diagnostic Report"
Usage: #example
* status = #final
* code.text = "Genomic Report"
* subject.reference = "Patient/example"
* media[0].extension[0].url = "http://example.org/fhir/StructureDefinition/diagnosticreport-media-documentreference"
* media[0].extension[0].valueReference.reference = "DocumentReference/genomic-report-pdf"

Instance: genomic-report-pdf
InstanceOf: DocumentReference
Title: "Genomic Report PDF"
Usage: #example
* status = #current
* content.attachment.url = "https://example.org/files/genomic.pdf"
* type.text = "PDF of genomic diagnostic report"
* context.related.reference = "DiagnosticReport/example-report"
```

---

## ⚠️ Circular Reference Caveat

If both the `DiagnosticReport` references a `DocumentReference` via this extension, **and** the `DocumentReference.context.related` references the same `DiagnosticReport`, this forms a **cyclic graph**.

Recommendations:
- Avoid circular references in practice to prevent infinite loops in traversals.
- Remove the `context.related` to DiagnosticReport in DocumentReference  if using this extension to prevent cycles.

RobertJCarroll commented:
Thanks for the detailed rundown. If it's helpful, we can discuss at the meeting on 12 August. I have a few key points I'd like to share in advance.

First: The main business requirement I'm advocating for is "Our model should describe what happened with an assay with semantic clarity consistent with FHIR." Given that your alternate proposal removes ServiceRequest entirely, I don't think I communicated that clearly when I said "We should be able to omit the request resource and the entire structure should still communicate what happened." I'm not advocating to remove ServiceRequest from the model, but indicating that DiagnosticReport or other resource must be able to capture the data about the assay. 

Second: I'm not seeing some of the limitations you are describing around Task. It does support using ActivityDefinitions through `Task.instantiatesCanonical`. It does support linking to participants and specimens through `Task.for` and `Task.focus` (or `Task.input` for more flexibility). It connects to the Requests through `Task.basedOn` and with other tasks through `Task.partOf`. I agree it doesn't fit well with other procedures, but in those cases we would use `Procedure`. 

We don't have to use `Task`- `DiagnosticReport` may be a fine fit with R5 in consideration, but I think it'd be helpful to narrow down the factors to just that element where we'd see a potential difference. My take on the breakdown is more like the following:
| Aspect                                         |  `DiagnosticReport` | `Task` |
| ----------------------------------- | --------------------- | ------ |
| 🔄 **Workflow Intent**                         | ✅ `DiagnosticReport.basedOn`       |    ✅ `Task.basedOn`                                  |
| 📄 **File Representation**                     | 🟡 `DiagnosticReport.media` does not align with existing DocRefs         | ✅ `Task.output` referencing `DocumentReference`                      |
| 🧫 **Specimen Linkage**                        | ✅ `DiagnosticReport.specimen`                 | 🟡 `Task.focus` or `Task.input` with slices |
| 🔁 **Reuse of Standard Workflows**             | 🚫 No link to reusable protocol definitions* | ✅ `Task.instantiatesCanonical`  |
| 🧬 **Support for Genomics & Beyond**           | 🟡 Genomics and Imaging studies only | 🟡 Generalizable, but at cost of semantic richness without additional task profiling                        |
| 🧰 **Support for Parameters, Runtime Context** | 🟡 Partial (mostly in Task, some in SR)                                                                 | ✅ Strong via `Task.input`/`output`                                       |
| 🧵 **Traceability Across Resources**           | ✅ Works fine             | ✅Allows connecting Tasks into wider groups                       |

*It's possible to signal the planned activity definition through the SR, but not the implemented activity. In R5 and assuming you have a compatible Study resource, it's there.

It's pretty close in my book- mostly coming down to the "squishy" Task that allows dumping more info in versus the more rigid DiagnosticReport that may need additional, restricted Study resources for details.

bwalsh commented:
Re. [R4 Task](https://hl7.org/fhir/R4/task.html)
Note that `focus` and `for` are scalar.   A common use case is for an assay to operate on a group of Specimens (e.g. gVCF)

<img width="683" height="100" alt="image" src="https://github.com/user-attachments/assets/ed4c9003-03db-439f-a4e4-4493f68fc68a" />


Also, `input` and `output` are not by default searchable.  Nor can a user chain to them via _include _revinclude.

<img width="1154" height="809" alt="image" src="https://github.com/user-attachments/assets/0684054c-8087-4487-a2be-e36d2ce22d3b" />

torstees committed:
🍱Add OBI valuesets for Assay types and Instruments

* Using OBI for common assay strategies

* Support for OBI ontologies

* Update to use OBI code for WGS

* Correct system for OBI

* Added example codes for platform and set those VS as preferred

* Updated examples to use OBI terms

bwalsh committed:
adds DiagnosticReport.media

RobertJCarroll commented:
I like the addition of the `DiagnosticReport.media` extension there. Makes sense in the R5 context we live in today.

Having the clear `DiagnosticReport.subject` and `DiagnosticReport.specimen` are also great (mostly better understanding on my end).

This is a wider scope than this conversation, but I was talking with Allison re the "how do we link files back to participants and samples"- I think we want to open that conversation up again. dbGaP team had some ideas, too, with the additional subject extension.

bwalsh commented:
@RobertJCarroll @torstees @teslajoy 
Saw this today, note similarity to assay approach https://pmc.ncbi.nlm.nih.gov/articles/PMC7080712/

<img width="668" height="502" alt="image" src="https://github.com/user-attachments/assets/d8d2de1b-3f29-47da-becf-9acfe1e39cd7" />
